src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -65,8 +66,12 @@ class SyncController(
         case (true, false) =>
           startRegularSync()
         case (false, false) =>
-          fastSyncStateStorage.purge()
-          startRegularSync()
+          //Check whether fast sync was started before
+          if (fastSyncStateStorage.getSyncState().isDefined) {
```

Comment:
Even if this is ok, it's confusing to me to retrieve if the fastSync has ever started doing `fastSyncStateStorage.getSyncState()` but to check if it has finished we need to do it based on  `appStateStorage.isFastSyncDone()`